src/clipboard_handlers/cell.ts

```diff
@@ -0,0 +1,289 @@
+import { CommandResult } from "..";
+import { formatValue } from "../helpers";
+import { pastedZones } from "../helpers/clipboard/clipboard_helpers";
+import {
+  CellPosition,
+  ClipboardCell,
+  ClipboardData,
+  ClipboardOptions,
+  ClipboardPasteTarget,
+  HeaderIndex,
+  UID,
+  Zone,
+} from "../types";
+import { ClipboardHandler } from "./abstract_clipboard_handler";
+
+interface ClipboardContent {
+  cells: ClipboardCell[][];
+  zones: Zone[];
+  sheetId: UID;
+  isCutOperation: boolean;
+}
+
+export class CellClipboardHandler extends ClipboardHandler<ClipboardContent> {
+  isCutAllowed(data: ClipboardData) {
+    if ("zones" in data && data.zones.length !== 1) {
+      return CommandResult.WrongCutSelection;
+    }
+    return CommandResult.Success;
+  }
+
+  copy(isCutOperation: boolean, data: ClipboardData): ClipboardContent | undefined {
+    if (!("zones" in data) || !data.zones.length) {
+      return;
+    }
+    const sheetId = this.getters.getActiveSheetId();
+    const zones = data.zones;
+    if (!zones.length) {
+      return {
+        cells: [[]],
+        zones: [],
+        isCutOperation,
```

Comment:
`isCutOperation` should probably be handled by the clipboard and given as argument to the paste, rather than each handler having to store it in their state